pkg/util/watch.go

```diff
@@ -407,10 +450,27 @@ type WatchStore[T any] struct {
 	objects map[types.UID]*T
 	mutex   sync.Mutex
 
+	triggerRelist chan struct{}
+	relisted      chan struct{}
+
 	stopSignal SignalSender
 	stopped    atomic.Bool
 }
 
+// Relist triggers re-listing the WatchStore, returning a channel that will be closed once the
+// re-list is complete
+func (w *WatchStore[T]) Relist() <-chan struct{} {
```

Comment:
this is ok, but you could also avoid the lock, and send (what is a reference, in effect) the response channel through the trigger channel, so `ch := make(chan struct{}); w.triggerRelist <- ch`, and then you wouldn't have to store the rellisted channel in the struct. 

Also the non-blocking send looks dodgy to my eyes, as I read the code, but I realized that it wasn't because you're mutating the w.relisted channel in the other thread.

Reply:
> this is ok, but you could also avoid the lock, and send (what is a reference, in effect) the response channel through the trigger channel, so ch := make(chan struct{}); w.triggerRelist <- ch, and then you wouldn't have to store the relisted channel in the struct.

I decided not to create a channel for each call to `Relist` because subjectively, it felt like there's more cases that could lead to an unreasonable number of channels getting created (part of the challenge with having `Relist` return a channel).

> Also the non-blocking send looks dodgy to my eyes, as I read the code, but I realized that it wasn't

Well, because of this I double-checked and realized the send _was_ dodgy :upside_down_face: Needed to have `w.triggerRelist` be non-blocking, fixed that now.

> mutating the w.relisted channel in the other thread

Yep. It's more complicated than I'd like, but getting the semantics I want is actually kind of hard.